core/src/main/kotlin/Library.kt

```diff
@@ -1,62 +1,164 @@
-fun parseVsq(file: File): Promise<Document> = parse(listOf(file), Format.Vsq)
+fun parseVsq(file: File): Promise<ProjectContainer> = parse(listOf(file), Format.Vsq)
 
 @JsExport
-fun parseVocaloidMid(file: File): Promise<Document> = parse(listOf(file), Format.VocaloidMid)
+fun parseVocaloidMid(file: File): Promise<ProjectContainer> = parse(listOf(file), Format.VocaloidMid)
 
 @JsExport
-fun parseStandardMid(file: File): Promise<Document> = parse(listOf(file), Format.StandardMid)
+fun parseStandardMid(file: File): Promise<ProjectContainer> = parse(listOf(file), Format.StandardMid)
 
 @JsExport
-fun parsePpsf(file: File): Promise<Document> = parse(listOf(file), Format.Ppsf)
+fun parsePpsf(file: File): Promise<ProjectContainer> = parse(listOf(file), Format.Ppsf)
 
 @JsExport
-fun parseUfData(files: List<File>): Promise<Document> = parse(files, Format.UfData)
+fun parseUfData(file: File): Promise<ProjectContainer> = parse(listOf(file), Format.UfData)
 
-private fun parse(files: List<File>, format: Format): Promise<Document> = GlobalScope.promise {
+private fun parse(files: List<File>, format: Format): Promise<ProjectContainer> = GlobalScope.promise {
     val project = format.parser(files, ImportParams())
-    UfData.generateDocument(project)
+    ProjectContainer(project)
 }
 
+@JsExport
+fun generateVsqx(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.Vsqx)
+
+@JsExport
+fun generateVpr(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.Vpr)
+
+@JsExport
+fun generateUstZip(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.Ust)
+
+@JsExport
+fun generateUstx(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.Ustx)
+
+@JsExport
+fun generateCcs(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.Ccs)
+
+@JsExport
+fun generateSvp(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.Svp)
+
+@JsExport
+fun generateS5p(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.S5p)
+
+@JsExport
+fun generateMusicXmlZip(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.MusicXml)
 
 @JsExport
-fun hello(): String = "core"
+fun generateDv(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.Dv)
+
+@JsExport
+fun generateVsq(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.Vsq)
+
+@JsExport
+fun generateVocaloidMid(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.VocaloidMid)
+
+@JsExport
+fun generateStandardMid(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.StandardMid)
+
+@JsExport
+fun generateUfData(project: ProjectContainer): Promise<ExportResult> = generate(project, Format.UfData)
+
+private fun generate(project: ProjectContainer, format: Format): Promise<ExportResult> = GlobalScope.promise {
+    format.generator(project.project, listOf())
```

Comment:
For now it's fine, but we can add supports for other features (e.g. pitch conversion) in the future